hazelcast/src/main/java/com/hazelcast/query/impl/predicates/PartitionPredicateImpl.java

```diff
@@ -121,19 +156,24 @@ public boolean equals(Object o) {
         if (o == null || getClass() != o.getClass()) {
             return false;
         }
-
         PartitionPredicateImpl<?, ?> that = (PartitionPredicateImpl<?, ?>) o;
-
-        if (partitionKey != null ? !partitionKey.equals(that.partitionKey) : that.partitionKey != null) {
-            return false;
-        }
-        return target != null ? target.equals(that.target) : that.target == null;
+        return Objects.equals(partitionKeys, that.partitionKeys)
+               && Objects.equals(target, that.target);
     }
 
     @Override
     public int hashCode() {
-        int result = partitionKey != null ? partitionKey.hashCode() : 0;
+        int result = partitionKeys != null ? partitionKeys.hashCode() : 0;
         result = 31 * result + (target != null ? target.hashCode() : 0);
         return result;
     }
+
+    private Object getSinglePartitionKey() {
+        for (Object key : partitionKeys) {
+            return key;
+        }
+        throw new RuntimeException(
+            "Unreachable branch. PartitionPredicateImpl constructor should check and throw if partitionKeys is empty"
+        );
+    }
```

Comment:
Maybe we can add a javaDoc for method's purpose and an assertion to catch unexpected cases.
```
    /**
     * This method is here only for backward compatibility
     * with versions < 5.2.
     * @return a single partition key.
     */
    private Object getSinglePartitionKey() {
        assert partitionKeys.size() == 1;
        
for (Object key : partitionKeys) {
            return key;
        }
        throw new RuntimeException(
            "Unreachable branch. PartitionPredicateImpl constructor should check and throw if partitionKeys is empty"
        );
    }
```

Reply:
JavaDoc for a `private` method seems a bit overkill